gms/gossiper.cc

```diff
@@ -166,6 +169,7 @@ void gossiper::do_sort(utils::chunked_vector<gossip_digest>& g_digest_list) cons
 // Depends on
 // - no external dependency
 future<> gossiper::handle_syn_msg(const rpc::client_info& cinfo, gossip_digest_syn syn_msg) {
+    co_await update_address_map(cinfo);
```

Comment:
@kbr-scylla but the generation comes from gossiper endpoint_state, and here we don't have that in the client_info.
I like @tgrabiec's idea of fencing off messages from stale peer.
The quesation is would it need adding connection establishment verbs or not.

Reply:
> The quesation is would it need adding connection establishment verbs or not.

It would fit into messaging_verb::CLIENT_ID, but it is no_wait, so we can't reuse it. We need to add SERVER_ID.

Reply:
SERVER_ID could also return the generation which we can use when populating the address mapping.

Reply:
> If there's an existing connection for that host_id, wouldn't it be better to disconnect it first before allowing a new connection?

There may be multiple connections (from different shards) coming from a single peer.

Reply:
> > If there's an existing connection for that host_id, wouldn't it be better to disconnect it first before allowing a new connection?
> 
> There may be multiple connections (from different shards) coming from a single peer.

OK, makes sense.
How about a stale connection with same host_id and different inet_address?
Is it even possible, or since it requires restart the previous connections are bound to be closed already?

Reply:
> How about a stale connection with same host_id and different inet_address?
> Is it even possible, or since it requires restart the previous connections are bound to be closed already?

I think it's possible if the old node did not close the connection gracefully. 

We could close these, but what guarantee would you like to achieve by that?

Reply:
I'd like to fence possibly in-flight updates from the previous incarnation.
That said, ordering by version/generation should make sure that older updates to the endpoint state are ignored.

Reply:
> > It would fit into messaging_verb::CLIENT_ID, but it is no_wait, so we can't reuse it.
> 
> Do we need the establisher of connection (sender) to receive a response?

For host_id-based routing yes, because we want to know which host we connected to.